questionnaire/questionnaire-dal.js

```diff
@@ -62,10 +62,69 @@ function questionnaireDAL(spec) {
         return questionnaire;
     }
 
+    async function getQuestionnaireSubmissionStatus(questionnaireId) {
+        let submissionStatus;
+
+        try {
+            submissionStatus = await db.query(
+                'SELECT submission_status FROM questionnaire WHERE id = $1',
+                [questionnaireId]
+            );
+
+            if (submissionStatus.rows.length === 0) {
+                // No instance was found
+                throw new VError(
+                    {
+                        name: 'ResourceNotFound'
+                    },
+                    `Questionnaire "${questionnaireId}" not found`
+                );
+            }
+        } catch (err) {
+            throw err;
+        }
+
+        return submissionStatus;
+    }
+
+    async function updateQuestionnaireSubmissionStatus(questionnaireId, submissionStatus) {
+        let result;
+
+        try {
+            result = await db.query(
```

Comment:
Should that be done here or should this check be done at the service layer using the getQuestionnaireSubmissionStatus function?

Reply:
I think given that you're supplying a specific id to be updated, then if it isn't updated, it should throw an exception here. If the service layer need to do anything with it it can. I think in one of my previous commits I've handled it at the service layer but should have been here.

As discussed, I also retuned the pg object as-is to the service layer, in future we might want to wrap it first.